src/Microsoft.Azure.CosmosRepository/IQueryResult.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Azure.CosmosRepository
+{
+    /// <summary>
+    /// Represent a full set of data from a cosmos query
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public interface IQueryResult<out T> where T : IItem
```

Comment:
We tend to use the generic argument name `TItem` on all our other geneirc implementations

Reply:
Ok will rename everywhere. Hopefully I'll get all place

> Hi, @filipmhpersson I have spoken with @IEvangelist and we have both agreed there is a great PR and we are happy for it to be merged.
> 
> We have some testing we'd like to do and some styling nits we will amend but other than we'd like to thank you for taking the time to raise this PR. If you are happy we can merge this and then get it released in the next few days :)
> 
> There are just some merge conflicts that need resolving first.

Awesome, I started yesterday with some styling (reming T to TItem where relevant). What else do you need from me? Should I resolve the merge conflicts and commit the styling I have done so far and you'll do what you need after that?

I have ideas on what could be done better but I think think this is a good enough version to get out there and see how it will work for developers. So it's probably better to just get this out and do improvements depending on feedback.